service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractMapper.java

```
@@ -0,0 +1,85 @@
+
+@Component
+public class EhrExtractMapper {
+    private static final Mustache EHR_EXTRACT_TEMPLATE = TemplateUtils.loadTemplate("ehr_extract_template.mustache");
+    private static final String SLASH = "/";
+
+    private final FhirParseService fhirParseService;
+    private final TaskIdService taskIdService;
+
+    @Autowired
+    public EhrExtractMapper(FhirParseService fhirParseService, TaskIdService taskIdService) {
+        this.fhirParseService = fhirParseService;
+        this.taskIdService = taskIdService;
+    }
+
+    public EhrExtractTemplateParameters mapJsonToEhrFhirExtractParams(EhrExtractStatus ehrExtractStatus,
+            GetGpcStructuredTaskDefinition getGpcStructuredTaskDefinition,
+            String json) {
+        Bundle bundle = fhirParseService.parseResource(json, Bundle.class);
+
+        return prepareEhrFhirExtractParamsFromFhirBundle(ehrExtractStatus, getGpcStructuredTaskDefinition, bundle);
+    }
+
+    public String mapEhrExtractToXml(EhrExtractTemplateParameters ehrExtractTemplateParameters) {
+        return TemplateUtils.fillTemplate(EHR_EXTRACT_TEMPLATE, ehrExtractTemplateParameters);
+    }
+
+    private EhrExtractTemplateParameters prepareEhrFhirExtractParamsFromFhirBundle(EhrExtractStatus ehrExtractStatus,
+            GetGpcStructuredTaskDefinition getGpcStructuredTaskDefinition,
+            Bundle bundle) {
+        EhrExtractTemplateParameters ehrExtractTemplateParameters = new EhrExtractTemplateParameters();
+        ehrExtractTemplateParameters.setEhrFolderId(taskIdService.createNewTaskId());
+        ehrExtractTemplateParameters.setPatientId(ehrExtractStatus.getEhrRequest().getNhsNumber());
+        ehrExtractTemplateParameters.setRequestId(ehrExtractStatus.getEhrRequest().getRequestId());
+        ehrExtractTemplateParameters.setEhrExtractId(getGpcStructuredTaskDefinition.getConversationId());
+
+        Optional<Patient> extractedPatient = extractPatientFromBundle(bundle);
```

This value is treated as optional. Is it required for the mapping? If it is required we need to throw an error if it is missing.

---

Yeah it is the entire Patient resource. If that's not in the structured record we have a serious problem. Should throw an error if it is missing. Probably throw from within `extractPatientFromBundle`? I suspect this method will get reused in future mappings

---

Ok